plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/internal/CloudSdkInstallJob.java

```diff
@@ -48,8 +63,51 @@ public boolean belongsTo(Object family) {
 
   @Override
   protected IStatus run(IProgressMonitor monitor) {
-    // TODO(chanseok): to be implemented: https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/2753
-    consoleStream.println("to be implemented");
-    return Status.OK_STATUS;
+    if (monitor.isCanceled()) {
+      return Status.CANCEL_STATUS;
+    }
+
+    try {
+      ManagedCloudSdk managedSdk = getManagedCloudSdk();
+      if (!managedSdk.isInstalled()) {
+        SdkInstaller installer = managedSdk.newInstaller();
+        installer.install(new MessageConsoleWriterListener(consoleStream));
+      }
+
+      if (!managedSdk.hasComponent(SdkComponent.APP_ENGINE_JAVA)) {
+        SdkComponentInstaller componentInstaller = managedSdk.newComponentInstaller();
+        componentInstaller.installComponent(
+            SdkComponent.APP_ENGINE_JAVA, new MessageConsoleWriterListener(consoleStream));
+      }
+      return Status.OK_STATUS;
+
+    } catch (InterruptedException | ClosedByInterruptException e) {
+      return Status.CANCEL_STATUS;
+    } catch (IOException | ManagedSdkVerificationException | SdkInstallerException |
+        CommandExecutionException | CommandExitException e) {
+      return StatusUtil.error(this, "Failed to install the Google Cloud SDK.", e);
+
+    } catch (UnsupportedOsException e) {
+      throw new RuntimeException("Cloud Tools for Eclipse supports Windows, Linux, and Mac only.");
```

Comment:
Would it be better to return a StatusUtil.error in these cases rather than throwing a runtime exception?

Reply:
Seems reasonable to me.  The message isn't quite right — the managed cloud installation only supports Windows, Linux, and Mac.